### PR TITLE
emit: optimize Bool GAS cost

### PR DESCRIPTION
**Pull request by roman-khimov**
NOT is 1 byte shorter and 2048 times cheaper than CONVERT. Inspired by neo-project/neo-vm#493.

### CODE REVIEW

**Review by AnnaShaleva — COMMENTED** [review]
We probably need to regenerate our testchain and update RPC server test data (test contracts hashes and AVM bytes) wrt this change.

**Comment by roman-khimov** [comment]
Why? I had a thought on RPC test data, but this would be an unnecessary change to me, old NEFs still work fine. And this new code is tested in a lot of places including contracts deployed by CLI tests.

File: pkg/vm/emit/emit.go [hunk 1]
```diff
@@ -31,12 +31,11 @@ func Opcodes(w *io.BinWriter, ops ...opcode.Opcode) {
...
-	if ok {
-		Opcodes(w, opcode.PUSHT)
-	} else {
-		Opcodes(w, opcode.PUSHF)
+	var opVal = opcode.PUSHF
+	if !ok {
+		opVal = opcode.PUSHT
 	}
-	Instruction(w, opcode.CONVERT, []byte{byte(stackitem.BooleanT)})
+	Opcodes(w, opVal, opcode.NOT)
```
Comment (fyrchik) [inline pkg/vm/emit/emit.go:38]: I am wondering whether we can do `NOT+NOT` in compiler.

**Review by fyrchik — APPROVED** [review]

File: pkg/vm/emit/emit.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/vm/emit/emit.go:38]: Huh, I've also looked there, can be done, but I've thought that it'd seriously complicate generic conversion code which will affect every type conversion while only benefit one. Looking at it once again I see we can emit type-specific conversion easily, so let me try doing that.

Comment (roman-khimov) [reply pkg/vm/emit/emit.go:38]: Added this one also, unfortunately it can't be extended to integers because of neo-project/neo-vm#495.

**Comment by roman-khimov** [comment]
> then other developer will have to do this

Yeah, that's the intention! It's an RPC server test after all and RPC functions are not affected in any way by this change.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**